### PR TITLE
Google provider docstring improvements

### PR DESCRIPTION
**Pull request by uranusjr**
Related suggestions found when reviewing https://github.com/apache/airflow/pull/31422.

### CODE REVIEW

**Review by eladkal — APPROVED** [review]

**Review by vincbeck — APPROVED** [review]

**Comment by uranusjr** [comment]
Apparently I broke _something_ but I can’t figure out what’s going on. The error message is quite unhelpful too 😕

**Comment by potiuk** [comment]
> Apparently I broke _something_ but I can’t figure out what’s going on. The error message is quite unhelpful too confused

You can thank sphinx-autoapi  for those unhelpful messages, no-one can do much about it. 

```
 WARNING: Definition list ends without a blank line; unexpected unindent.
```

Apparently the docstring generated rst (in `_api` folder) contains invalid .rst after your change with missing blank like.

**Comment by potiuk** [comment]
`File path: apache-airflow-providers-google/_api/airflow/providers/google/cloud/hooks/bigquery/index.rst `

You can do `breeze build-docs --package-filter apache-airflow-providers-google` and look at the generated .rst to find out the iseue. Ths is as much as can be done there.

**Comment by uranusjr** [comment]
I did that, the problem is I can’t even figure out where the generated rst has any issues. The files just look good to me. 🤯

**Comment by potiuk** [comment]
> I did that, the problem is I can’t even figure out where the generated rst has any issues. The files just look good to me. exploding_head

Been there a few times. What worked for me before is doing bisecting and removing parts of the change and figuring out which one is causing it... Not ideal, but works eventually.

And there are hopes we might get some better error messages - I see some activity happening on https://github.com/readthedocs/sphinx_rtd_theme/issues/1463 where people started to complain big time on sphinx_rtd_theme not being updated to Sphinx 7 and THIS PR https://github.com/readthedocs/sphinx_rtd_theme/pull/1464  has just gotten green, so maybe we should also chime in there ? 

I think we  really need it.

Let me take a look maybe I can figure that one out quickly.

**Comment by potiuk** [comment]
BTW. `breeze build-docs --package-filter apache-airflow-providers-googe --one-pass-only --docs-ony` will give you the fastest iteration on this (but the fact that google is about 50% of all provider's code, does not make it exactly fast

File: airflow/providers/google/cloud/hooks/bigquery.py [hunk 1]
```diff
@@ -618,10 +610,10 @@ def create_external_table(
         :param labels: A dictionary containing labels for the BiqQuery table.
         :param description: A string containing the description for the BigQuery table.
         :param encryption_configuration: [Optional] Custom encryption configuration (e.g., Cloud KMS keys).
-            **Example**: ::
+            .. code-block:: python
```
Comment (potiuk) [inline airflow/providers/google/cloud/hooks/bigquery.py:616]: Result of bisecting - you miss empty line here. Also indenting is not consistent with other places but it seem to not cause the problem.

File: airflow/providers/google/cloud/hooks/bigquery.py [hunk 2]
```diff
@@ -1716,10 +1690,10 @@ def run_load(
             by one or more columns. BigQuery supports clustering for both partitioned and
             non-partitioned tables. The order of columns given determines the sort order.
         :param encryption_configuration: [Optional] Custom encryption configuration (e.g., Cloud KMS keys).
-            **Example**: ::
+            .. code-block:: python
```
Comment (potiuk) [inline airflow/providers/google/cloud/hooks/bigquery.py:1705]: This is the second place where empty line before is missing.

**Comment by potiuk** [comment]
Created https://github.com/apache/airflow/issues/31963

**Comment by potiuk** [comment]
Added comment here as well - https://github.com/readthedocs/sphinx_rtd_theme/issues/1463#issuecomment-1594855212 @uranusjr -> I'd appreciate some support/words of encouragement for sphinx_rtd_them developers.

I **really** feel your pain BTW.

**Comment by uranusjr** [comment]
FWIW we really should switch away from sphinx-autoapi as well (in favor of `sphinx.ext.autodoc`), I think that’s another blocker for Sphinx 7 and generally a pain to interact with.

**Review by phanikumv — APPROVED** [review]

**Review by jedcunningham — APPROVED** [review]

**Pull request merged**